internal/decryption/keyproviders/local_key.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keyproviders
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"encoding/base64"
+)
+
+func decryptWithLocalKey(getLocalKeyFn func() (string, error), encryptedLEK, iv []byte) ([]byte, error) {
```

Comment:
avoid function pointers, try polymorphism instead

Reply:
changed to Decrypt func and created an interface, please let me know if you think this can be improved
type KeyProvider interface {
	DecryptKey(encryptedLEK []byte, iv []byte) ([]byte, error)
}